library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java

```diff
@@ -127,15 +148,17 @@ private void drawCircle(Canvas canvas) {
      * @param canvas canvas to draw
      */
     private void drawRoundedRectangle(Canvas canvas) {
-        float width = mCalculator.roundRectWidth();
         float left = mCalculator.roundRectLeft();
         float top = mCalculator.roundRectTop(mAnimCounter, mAnimMoveFactor);
         float right = mCalculator.roundRectRight();
         float bottom = mCalculator.roundRectBottom(mAnimCounter, mAnimMoveFactor);
-        canvas.drawRect(left, top, right, bottom, mErasePaint);
-        canvas.drawCircle(left, mCalculator.getCircleCenterY(),
-                mCalculator.roundRectLeftCircleRadius(mAnimCounter, mAnimMoveFactor), mErasePaint);
-        canvas.drawCircle(left + width, mCalculator.getCircleCenterY(),
-                mCalculator.roundRectLeftCircleRadius(mAnimCounter, mAnimMoveFactor), mErasePaint);
+
+        RectF rectf = new RectF(left, top, right, bottom);
```

Comment:
You should avoid allocating objects during a drawing or layout operation. These are called frequently, so a smooth UI can be interrupted by garbage collection pauses caused by the object allocations.
`new RectF()` should not be used here. Because of this I used a different way. (2 circles and a rectangle). 
Can you draw a rounded rectangle in a different way? (Without object allocations each time)